modules/shortcodes/vimeo.php

```diff
@@ -172,6 +172,77 @@ function vimeo_shortcode( $atts ) {
 
 add_shortcode( 'vimeo', 'vimeo_shortcode' );
 
+/**
+ * Filters the Vimeo shortcode to be AMP-compatible.
+ *
+ * @param string $html The shortcode HTML.
+ * @param string $shortcode_tag The shortcode's tag (name).
+ * @param array  $attr The attributes of the shortcode.
+ * @return string The filtered HTML.
+ */
+function amp_vimeo_shortcode( $html, $shortcode_tag, $attr ) {
+	if (
+		( class_exists( 'Jetpack_AMP_Support' ) && ! Jetpack_AMP_Support::is_amp_request() )
+		||
+		'vimeo' !== $shortcode_tag
+	) {
+		return $html;
+	}
+
+	$video_id = null;
+	if ( isset( $attr[0] ) ) {
+		$video_id = jetpack_shortcode_get_vimeo_id( $attr );
+	} elseif ( isset( $attr['id'] ) ) {
+		$video_id = $attr['id'];
+	}
+
+	list( $width, $height ) = jetpack_get_amp_vimeo_dimensions( $attr );
+	return jetpack_render_amp_vimeo( compact( 'video_id', 'width', 'height' ) );
+}
+
+/**
+ * Gets the width and height of the AMP [vimeo] shortcode.
+ *
+ * @param array $attr The shortcode attributes.
+ * @return array The dimensions, width at index 0, and height at index 1.
+ */
+function jetpack_get_amp_vimeo_dimensions( $attr ) {
+	$default_width  = 600;
+	$default_height = 338;
+	$aspect_ratio   = $default_height / $default_width;
+
+	if ( ! empty( $GLOBALS['content_width'] ) ) {
+		$width  = $GLOBALS['content_width'];
+		$height = round( $GLOBALS['content_width'] * $aspect_ratio );
+	} else {
+		$width  = isset( $attr['width'] ) ? $attr['width'] : $default_width;
+		$height = isset( $attr['height'] ) ? $attr['height'] : $default_height;
+	}
+
+	return array( $width, $height );
```

Comment:
Do we want to run those values through `absint` here?

Reply:
Good idea, https://github.com/Automattic/jetpack/pull/13921/commits/0a6732aa9d5c14f472e2033a5ea8e54c8a28a6a8 applies it